### DISCUSSION TITLE
RuntimeError: cuDNN error: CUDNN_STATUS_NOT_INITIALIZED

### DISCUSSION THREAD

**Discussion by openmindead** [openai/whisper#909] [Ideas]
I've got a problem using whisper on a quite weak and outdated Nvidia MX150 with 2 Gb of memory, and that problem is identical to this StackOverflow issue: https://stackoverflow.com/questions/66588715/runtimeerror-cudnn-error-cudnn-status-not-initialized-using-pytorch
I found out that the top-rated SO solution has resolved the issue but I am not sure that this approach is suitable for your project, my goal just is to make you aware of it. In brief, I made these changes exactly as described on the above mentioned page:
```
--- whisper_old 2023-01-30 00:09:59.635205494 +1000
+++ whisper     2023-01-29 18:21:51.905253104 +1000
@@ -2,7 +2,17 @@
 # -*- coding: utf-8 -*-
 import re
 import sys
+import torch
 from whisper.transcribe import cli
+
+
+def force_cudnn_initialization():
+    s = 32
+    dev = torch.device('cuda')
+    torch.nn.functional.conv2d(torch.zeros(s, s, s, s, device=dev), torch.zeros(s, s, s, s, device=dev))
+
+
 if __name__ == '__main__':
+    force_cudnn_initialization()
     sys.argv[0] = re.sub(r'(-script\.pyw|\.exe)?$', '', sys.argv[0])
     sys.exit(cli())
```
And whisper stopped to fail on each launch with `RuntimeError: cuDNN error: CUDNN_STATUS_NOT_INITIALIZED` error.

There might be a better and more clean solution to this, hope you'll get to it.
PS: this is not "out of memory" thing. I ran whisper using small model, 2 Gb is enough for that.

Thanks for your time.

**Comment by akhiilkasare** [comment]
Delete the current installed version and run the below command.

pip3 install torch==1.13.1+cu116 --extra-index-url https://download.pytorch.org/whl/cu116

**Reply by openmindead** [reply]
Yeah, I should have tried this of course (and will do), it really could have something to do with default packages provided by Arch Linux (torch is 1.13.1, but cuda is 11.8).
Thanks for the tip!

UPD: Well, unfortunately it didn't work and made it worse, whisper wrapper still fails with CUDNN_STATUS_NOT_INITIALIZED error, and trying to circumvent it with forcing cuda init which worked before now fails with "out of memory"-ish error during loading the same small model using interactive console.
Anyway thanks for having a look at it.
